pgbouncer_bootstrapper.py

```diff
@@ -0,0 +1,134 @@
+    """Patch the existing Kubernetes secret for PostgreSQL."""
+    metadata = client.V1ObjectMeta(labels={'component': secret_name})
+
+    body = client.V1Secret(
+        api_version="v1",
+        kind="Secret",
+        metadata=metadata,
+        string_data={'pgbouncer.ini': pgbouncer_ini, 'users.txt': users_txt})
+
+    try:
+        kube.patch_namespaced_secret(secret_name, namespace, body)
+        click.echo("Successfully patched secret")
+    except Exception as e:
+        print(e)
+        click.echo("Error patching secret")
+
+
+def ensure_conn_secret(kube, namespace, secret_name, pgbouncer_ini, users_txt):
+    """Create/update Kubernetes secret for PostgreSQL."""
+    # Search for Secret
+    kwargs = dict(label_selector=f'component={secret_name}', limit=1)
+
+    try:
+        secrets = kube.list_namespaced_secret(namespace, **kwargs)
+        if len(secrets.items) != 1:
+            click.echo("Secret does not exist, creating...")
+            create_pgbouncer_secret(kube, namespace, secret_name, pgbouncer_ini, users_txt)
+        else:
+            click.echo("Secret exists, patching...")
+            patch_pgbouncer_secret(kube, namespace, secret_name, pgbouncer_ini, users_txt)
+
+    except Exception as e:
+        click.echo(e)
+
+
+@click.command()
+@click.option('--bootstrap-db', envvar='BOOTSTRAP_DB', required=True)
+@click.option('--release-name', envvar='RELEASE_NAME', required=True)
+@click.option('--pool-size', envvar='PGBOUNCER_POOL_SIZE', required=True)
+@click.option('--pgbouncer-port', envvar='PGBOUNCER_PORT', required=True)
+@click.option('--max-client-conn', envvar='MAX_CLIENT_CONN', required=True)
+@click.option('--secret-name', envvar='SECRET_NAME', required=True)
+@click.option('--namespace', envvar='NAMESPACE', required=True)
+@click.option('--in-cluster', envvar='IN_CLUSTER', type=bool, default=False)
+def main(bootstrap_db, release_name, pool_size, pgbouncer_port, max_client_conn, secret_name, namespace, in_cluster):
+    kube_client = create_kube_client(in_cluster)
+    parsed_uri = urlparse(bootstrap_db)
+    database = parsed_uri.path[1:]
+    hostname = parsed_uri.hostname
+
+    try:
+        sslmode = parse_qs(parsed_uri.query, True)['sslmode']
+    except KeyError:
+        # I don't care about security, and I don't want to pay the overhead of encryption.
+        sslmode = 'disable'
+
+    # ensure the k8 secret exists for pgbouncer
+    pgbouncer_ini = PGBOUNCER_CONFIG.format(
+        release_name=release_name,
+        host=hostname,
+        db=database,
+        user=parsed_uri.username,
+        port=parsed_uri.port,
+        max_client_conn=max_client_conn,
+        sslmode=sslmode,
+        pool_size=pool_size,
+        pgbouncer_port=pgbouncer_port,
+    )
+    users_txt = '"{}" "{}"'.format(parsed_uri.username, parsed_uri.password)
+    ensure_conn_secret(kube_client, namespace, secret_name, pgbouncer_ini, users_txt)
```

Comment:
Slight change to API:
```suggestion
    ensure_conn_secret(kube_client, namespace, secret_name, {'pgbouncer.ini': pgbouncer_ini, 'users.txt': users_txt})
```

would be easier to follow/less special casing here (then make the chain of functions just take the `string_data` argument instead of the two values)